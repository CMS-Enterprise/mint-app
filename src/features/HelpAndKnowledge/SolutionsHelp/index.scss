.operational-solution-help {
    &__category {
        .usa-card__container {
            border-radius: 4px !important;
            padding-top: 1rem !important;
            padding-bottom: 1rem !important;
            height: 7rem !important;
            display: flex !important;
            justify-content: space-between !important;
        }
    }
}
