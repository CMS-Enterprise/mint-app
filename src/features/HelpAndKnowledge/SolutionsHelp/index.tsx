/*
Operational Solution portion of the MINT Help and Knowledge Center
Contains components for search, categories, and solutions cards
*/

import React, { useEffect, useMemo, useState } from 'react';
import { useLocation, useNavigate } from 'react-router-dom';
import { GridContainer } from '@trussworks/react-uswds';
import classNames from 'classnames';
import { MtoCommonSolutionSubject } from 'gql/generated/graphql';

import Divider from 'components/Divider';
import PageLoading from 'components/PageLoading';
import useHelpSolution from 'hooks/useHelpSolutions';
import useModalSolutionState from 'hooks/useModalSolutionState';

import CategoryFooter from './_components/CategoryFooter';
import SolutionHelpCardGroup from './_components/SolutionHelpCardGroup';
import SolutionsHeader from './_components/SolutionsHeader';
import SolutionDetailsModal from './SolutionDetails/Modal';
import {
  HelpSolutionBaseType,
  HelpSolutionType,
  routeToEnumMap
} from './solutionsMap';

type OperationalSolutionsHelpProps = {
  className?: string;
};

// Return all solutions relevant to the current cateory
export const findCategoryMapByRouteParam = (
  categoryKey: MtoCommonSolutionSubject,
  solutions: HelpSolutionType[]
): HelpSolutionBaseType[] => {
  return solutions.filter(solution => {
    return solution.categories.includes(categoryKey);
  });
};

export const findSolutionByRouteParam = (
  route: string | null | undefined,
  solutions: HelpSolutionType[],
  isEnum: boolean = false
): HelpSolutionType | undefined => {
  if (!route) return undefined;
  const routeParam = isEnum ? route : routeToEnumMap[route];
  return [...solutions].find(solution => solution.key === routeParam);
};

// Query function to return solutions for matching name and acronym
export const searchSolutions = (
  query: string,
  solutions: HelpSolutionType[]
): HelpSolutionType[] => {
  const queryRegex = new RegExp(query.toLowerCase());
  return solutions.filter(
    solution =>
      solution.name.toLowerCase().includes(query.toLowerCase()) ||
      solution?.acronym?.toLowerCase().includes(query.toLowerCase()) ||
      solution.categories.some(e => queryRegex.test(e.replace('-', ' ')))
  );
};

const SolutionsHelp = ({ className }: OperationalSolutionsHelpProps) => {
  const location = useLocation();
  const navigate = useNavigate();

  const params = useMemo(
    () => new URLSearchParams(location.search),
    [location.search]
  );

  const category = params.get('category') as MtoCommonSolutionSubject;
  const page = params.get('page');
  const solutionEnumParam = params.get('solution-key');
  const modal = params.get('solution') || solutionEnumParam;

  if (!page) {
    params.set('page', '1');
    navigate(
      {
        search: params.toString()
      },
      {
        replace: true
      }
    );
  }

  const { helpSolutions, loading } = useHelpSolution();

  // Get the solution map details from solution route param
  const { prevPathname, selectedSolution } = useModalSolutionState();

  const [query, setQuery] = useState<string>('');
  const [resultsNum, setResultsNum] = useState<number>(0);

  const [querySolutions, setQuerySolutions] =
    useState<HelpSolutionBaseType[]>(helpSolutions);

  const fromModal: boolean =
    prevPathname.includes('solution=') ||
    prevPathname.includes('solution-key=');

  // Resets the query on route or category change
  // Also preserves the query/scroll when the modal is open/closed
  useEffect(() => {
    if (!page && location.pathname && (!category || (!modal && !fromModal))) {
      setQuery('');
      window.scrollTo(0, 0);
    }
    if (!query && !modal && !fromModal) {
      window.scrollTo(0, 0);
    }
  }, [page, location.pathname, category, modal, query, fromModal]);

  //  If no query, return all solutions, otherwise, matching query solutions
  useEffect(() => {
    if (query.trim()) {
      const paramsChange = new URLSearchParams(location.search);
      paramsChange.set('page', '1');
      navigate({ search: paramsChange.toString() });
      setQuerySolutions(searchSolutions(query, helpSolutions));
    } else {
      setQuerySolutions(helpSolutions);
    }
  }, [query, selectedSolution, helpSolutions, location.search, navigate]);

  // If viewing by category, render those solutions, otherwise render querySolutions
  const solutions = !category
    ? querySolutions
    : findCategoryMapByRouteParam(category, helpSolutions);

  if (loading) {
    return <PageLoading />;
  }

  return (
    <div className={classNames(className)}>
      {selectedSolution && (
        <SolutionDetailsModal
          solution={selectedSolution}
          openedFrom={prevPathname}
          closeRoute="/help-and-knowledge/operational-solutions"
        />
      )}

      <SolutionsHeader
        category={category}
        resultsNum={resultsNum}
        resultsMax={solutions.length}
        setQuery={setQuery}
        query={query}
      />

      <SolutionHelpCardGroup
        solutions={solutions}
        setResultsNum={setResultsNum}
        category={category}
      />

      <GridContainer className="margin-top-4">
        <Divider className="margin-top-6" />

        <CategoryFooter currentCategory={category} />
      </GridContainer>
    </div>
  );
};

export default SolutionsHelp;
