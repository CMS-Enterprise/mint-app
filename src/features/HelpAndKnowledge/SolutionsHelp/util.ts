import { tObject } from 'utils/translation';

import { AboutConfigType } from './SolutionDetails/Solutions/Generic/About';
import { TimelineConfigType } from './SolutionDetails/Solutions/Generic/Timeline';

export const timelineTranslationUtil = (solutionKey: string) =>
  tObject<keyof TimelineConfigType, any>(
    `helpAndKnowledge:solutions.${solutionKey}.timeline`
  );

export const aboutTranslationUtil = (solutionKey: string) =>
  tObject<keyof AboutConfigType, any>(
    `helpAndKnowledge:solutions.${solutionKey}.about`
  );
