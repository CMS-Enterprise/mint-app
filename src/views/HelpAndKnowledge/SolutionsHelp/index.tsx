/*
Operational Solution portion of the MINT Help and Knowledge Center
Contains components for search, categories, and solutions cards
*/

import React, { useEffect, useState } from 'react';
import { useLocation } from 'react-router-dom';
import { GridContainer } from '@trussworks/react-uswds';
import classNames from 'classnames';

import Divider from 'components/shared/Divider';
import OperationalSolutionCategories from 'data/operationalSolutionCategories';
import usePrevLocation from 'hooks/usePrevious';

import CategoryFooter from './_components/CategoryFooter';
import SolutionHelpCardGroup from './_components/SolutionHelpCardGroup';
import SolutionsHeader from './_components/SolutionsHeader';
import SolutionDetailsModal from './SolutionDetails/Modal';
import {
  helpSolutions,
  HelpSolutionType,
  operationalSolutionCategoryMap
} from './solutionsMap';

type OperationalSolutionsHelpProps = {
  className?: string;
};

// Return all solutions relevant to the current cateory
export const findCategoryMapByRoute = (
  route: string,
  solutions: HelpSolutionType[]
): HelpSolutionType[] => {
  const categoryKey: OperationalSolutionCategories | undefined =
    operationalSolutionCategoryMap[route];

  if (!categoryKey) return [];

  return [...solutions].filter(solution => {
    return solution.categories.includes(
      categoryKey as OperationalSolutionCategories
    );
  });
};

export const findSolutionByRoute = (
  route: string | null,
  solutions: HelpSolutionType[]
): HelpSolutionType | undefined => {
  if (!route) return undefined;
  return [...solutions].find(solution => solution.route === route);
};

// Query function to return solutions for matching name and acronym
export const searchSolutions = (
  query: string,
  solutions: HelpSolutionType[]
): HelpSolutionType[] => {
  return solutions.filter(
    solution =>
      solution.name.toLowerCase().includes(query.toLowerCase()) ||
      solution?.acronym?.toLowerCase().includes(query.toLowerCase())
  );
};

const SolutionsHelp = ({ className }: OperationalSolutionsHelpProps) => {
  const location = useLocation();

  const params = new URLSearchParams(location.search);

  const category = params.get('category');
  const solution = params.get('solution');
  const page = params.get('page');

  const prevLocation = usePrevLocation(location);
  const prevPathname = prevLocation?.pathname + (prevLocation?.search || '');

  const [query, setQuery] = useState<string>('');
  const [resultsNum, setResultsNum] = useState<number>(0);

  const [querySolutions, setQuerySolutions] = useState<HelpSolutionType[]>(
    helpSolutions
  );

  // Resets the query on route or category change
  // Also preserves the query/scroll when the modal is open/closed
  useEffect(() => {
    if (!page && location.pathname) {
      setQuery('');
      window.scrollTo(0, 0);
    }
  }, [page, location.pathname, category]);

  //  If no query, return all solutions, otherwise, matching query solutions
  useEffect(() => {
    if (query.trim()) {
      setQuerySolutions(searchSolutions(query, helpSolutions));
    } else {
      setQuerySolutions(helpSolutions);
    }
  }, [query, solution]);

  // If viewing by category, render those solutions, otherwise render querySolutions
  const solutions = !category
    ? querySolutions
    : findCategoryMapByRoute(category, helpSolutions);

  // Solution to render in modal
  const selectedSolution = findSolutionByRoute(solution, helpSolutions);

  return (
    <div className={classNames(className)}>
      {selectedSolution && (
        <SolutionDetailsModal
          solution={selectedSolution}
          openedFrom={prevPathname}
        />
      )}

      <SolutionsHeader
        category={category}
        resultsNum={resultsNum}
        resultsMax={solutions.length}
        setQuery={setQuery}
        query={query}
      />

      <SolutionHelpCardGroup
        solutions={solutions}
        setResultsNum={setResultsNum}
        category={category}
      />

      <GridContainer className="margin-top-4">
        <Divider className="margin-top-6" />

        <CategoryFooter />
      </GridContainer>
    </div>
  );
};

export default SolutionsHelp;
