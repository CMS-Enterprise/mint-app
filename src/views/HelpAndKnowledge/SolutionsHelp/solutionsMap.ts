import OperationalSolutionCategories from 'data/operationalSolutionCategories';
import { OperationalSolutionKey } from 'types/graphql-global-types';

type CategoryPropertiesType = {
  route: string; // route to display relevant categories
};

type CategoryType = {
  [key in OperationalSolutionCategories]: CategoryPropertiesType;
};

// Operational Solution categories
export const operationalSolutionCategoryMap: CategoryType = {
  applications: {
    route: 'applications'
  },
  communicationTools: {
    route: 'communication-tools'
  },
  contractorsContractVehicles: {
    route: 'contractors-and-contract-vehicles'
  },
  databaseDataManagement: {
    route: 'database-and-data-management'
  },
  dataExchange: {
    route: 'data-exchange'
  },
  dataReporting: {
    route: 'data-reporting'
  },
  helpDesks: {
    route: 'help-desks'
  },
  learning: {
    route: 'learning'
  },
  legal: {
    route: 'legal'
  },
  medicareAdvantagePartD: {
    route: 'medicare-advantage-and-part-d'
  },
  medicareFeeForService: {
    route: 'medicare-fee-for-service'
  },
  quality: {
    route: 'quality'
  }
};

export type HelpSolutionType = {
  enum: OperationalSolutionKey | null; // TODO: should not be null, some enums havent been created
  key: string; // used for translations
  route: string;
  categories: OperationalSolutionCategories[];
  acronym: string | null;
  name: string;
  pocName: string;
};

export const helpSolutions: HelpSolutionType[] = [
  {
    enum: OperationalSolutionKey.INNOVATION,
    key: 'innovation',
    route: '4-innovation',
    categories: [OperationalSolutionCategories.DATA_REPORTING],
    acronym: '4i',
    name: '4innovation',
    pocName: '4i/ACO-OS Team'
  },
  {
    enum: OperationalSolutionKey.ACO_OS,
    key: 'accountableCare',
    route: 'accountable-care-organization',
    categories: [OperationalSolutionCategories.DATA_REPORTING],
    acronym: '',
    name: '',
    pocName: ''
  },
  {
    enum: OperationalSolutionKey.APPS,
    key: 'automatedPlanPayment',
    route: 'automated-plan-payment-system',
    categories: [OperationalSolutionCategories.MEDICARE_ADVANTAGE_D],
    acronym: '',
    name: '',
    pocName: ''
  },
  {
    enum: null,
    key: 'centralizedDataExhange',
    route: 'centralized-data-exchange',
    categories: [OperationalSolutionCategories.DATA_EXCHANGE],
    acronym: '',
    name: '',
    pocName: ''
  },
  {
    enum: OperationalSolutionKey.CCW,
    key: 'ccWarehouse',
    route: 'chronic-conditions-warehouse',
    categories: [OperationalSolutionCategories.DATABASE_MANAGEMENT],
    acronym: '',
    name: '',
    pocName: ''
  },
  {
    enum: null,
    key: 'cmsBox',
    route: 'cms-box',
    categories: [OperationalSolutionCategories.DATA_EXCHANGE],
    acronym: '',
    name: '',
    pocName: ''
  },
  {
    enum: null,
    key: 'cmsQualtrics',
    route: 'cms-qualtrics',
    categories: [OperationalSolutionCategories.DATA_REPORTING],
    acronym: '',
    name: '',
    pocName: ''
  },
  {
    enum: OperationalSolutionKey.CBOSC,
    key: 'cbosc',
    route: 'consolidated-business-operations-support-center',
    categories: [OperationalSolutionCategories.HELP_DESKS],
    acronym: '',
    name: '',
    pocName: ''
  },
  {
    enum: null,
    key: 'cpiVetting',
    route: 'cpi-vetting',
    categories: [OperationalSolutionCategories.DATA_EXCHANGE],
    acronym: '',
    name: '',
    pocName: ''
  },
  {
    enum: null,
    key: 'electronicFileTransfer',
    route: 'electronic-file-transfer',
    categories: [OperationalSolutionCategories.DATA_EXCHANGE],
    acronym: '',
    name: '',
    pocName: ''
  },
  {
    enum: null,
    key: 'expandedDataFeedback',
    route: 'expanded-data-feedback-reporting',
    categories: [OperationalSolutionCategories.DATA_REPORTING],
    acronym: '',
    name: '',
    pocName: ''
  },
  {
    enum: OperationalSolutionKey.GOVDELIVERY,
    key: 'govDelivery',
    route: 'gov-delivery',
    categories: [OperationalSolutionCategories.COMMUNICATION_TOOLS],
    acronym: '',
    name: '',
    pocName: ''
  },
  {
    enum: OperationalSolutionKey.GRANT_SOLUTIONS,
    key: 'grantSolutions',
    route: 'grant-solutions',
    categories: [
      OperationalSolutionCategories.APPLICATIONS,
      OperationalSolutionCategories.COMMUNICATION_TOOLS
    ],
    acronym: '',
    name: '',
    pocName: ''
  },
  {
    enum: OperationalSolutionKey.HIGLAS,
    key: 'higlas',
    route: 'healthcare-integrated-general-kedger-accounting-system',
    categories: [OperationalSolutionCategories.DATA_EXCHANGE],
    acronym: '',
    name: '',
    pocName: ''
  },
  {
    enum: OperationalSolutionKey.NEW_CMMI_PROCESS,
    key: 'healthDataReporting',
    route: 'health-data-reporting',
    categories: [OperationalSolutionCategories.QUALITY],
    acronym: '',
    name: '',
    pocName: ''
  },
  {
    enum: OperationalSolutionKey.HPMS,
    key: 'healthPlanManagement',
    route: 'health-plan-management-system',
    categories: [OperationalSolutionCategories.MEDICARE_ADVANTAGE_D],
    acronym: '',
    name: '',
    pocName: ''
  },
  {
    enum: OperationalSolutionKey.IPC,
    key: 'innovationPayment',
    route: 'innovation-payment-contract',
    categories: [OperationalSolutionCategories.CC_VEHICLES],
    acronym: '',
    name: '',
    pocName: ''
  },
  {
    enum: OperationalSolutionKey.IDR,
    key: 'integratedDataRepository',
    route: 'integrated-data-repository',
    categories: [OperationalSolutionCategories.DATABASE_MANAGEMENT],
    acronym: '',
    name: '',
    pocName: ''
  },
  {
    enum: null,
    key: 'learningAndDiffusion',
    route: 'learning-and-diffusion-group',
    categories: [OperationalSolutionCategories.LEARNING],
    acronym: '',
    name: '',
    pocName: ''
  },
  {
    enum: null,
    key: 'legalVertical',
    route: 'legal-vertical',
    categories: [OperationalSolutionCategories.LEGAL],
    acronym: '',
    name: '',
    pocName: ''
  },
  {
    enum: OperationalSolutionKey.MDM,
    key: 'masterDataManagement',
    route: 'master-data-management',
    categories: [OperationalSolutionCategories.DATABASE_MANAGEMENT],
    acronym: '',
    name: '',
    pocName: ''
  },
  {
    enum: null,
    key: 'measureInstrumentDS',
    route: 'measure-and-instrument-development-and-support',
    categories: [OperationalSolutionCategories.CC_VEHICLES],
    acronym: '',
    name: '',
    pocName: ''
  },
  {
    enum: OperationalSolutionKey.MARX,
    key: 'marx',
    route: 'medicare-advantage-prescription-drug-system',
    categories: [OperationalSolutionCategories.MEDICARE_ADVANTAGE_D],
    acronym: '',
    name: '',
    pocName: ''
  },
  {
    enum: OperationalSolutionKey.OUTLOOK_MAILBOX,
    key: 'outlookMailbox',
    route: 'outlook-mailbox',
    categories: [OperationalSolutionCategories.COMMUNICATION_TOOLS],
    acronym: '',
    name: '',
    pocName: ''
  },
  {
    enum: null,
    key: 'qualityVertical',
    route: 'quality-vertical',
    categories: [OperationalSolutionCategories.QUALITY],
    acronym: '',
    name: '',
    pocName: ''
  },
  {
    enum: OperationalSolutionKey.RMADA,
    key: 'rmada',
    route: 'research-measurement-assessment-design-and-analysis',
    categories: [OperationalSolutionCategories.CC_VEHICLES],
    acronym: '',
    name: '',
    pocName: ''
  },
  {
    enum: OperationalSolutionKey.ARS,
    key: 'ars',
    route: 'salesforce-application-review-and-scoring',
    categories: [OperationalSolutionCategories.APPLICATIONS],
    acronym: '',
    name: '',
    pocName: ''
  },
  {
    enum: OperationalSolutionKey.CONNECT,
    key: 'salesforceConnect',
    route: 'salesforce-connect',
    categories: [OperationalSolutionCategories.LEARNING],
    acronym: '',
    name: '',
    pocName: ''
  },
  {
    enum: null,
    key: 'salesforceLOI',
    route: 'salesforce-letter-of-intent',
    categories: [OperationalSolutionCategories.APPLICATIONS],
    acronym: '',
    name: '',
    pocName: ''
  },
  {
    enum: OperationalSolutionKey.SALESFORCE_PORTAL,
    key: 'salesforcePortal',
    route: 'salesforce-project-officer-support-tool-portal',
    categories: [OperationalSolutionCategories.COMMUNICATION_TOOLS],
    acronym: '',
    name: '',
    pocName: ''
  },
  {
    enum: OperationalSolutionKey.RFA,
    key: 'salesforceRequestApplication',
    route: 'salesforce-request-for-application',
    categories: [OperationalSolutionCategories.APPLICATIONS],
    acronym: '',
    name: '',
    pocName: ''
  },
  {
    enum: OperationalSolutionKey.SHARED_SYSTEMS,
    key: 'sharedSystems',
    route: 'shared-systems',
    categories: [OperationalSolutionCategories.MEDICARE_FFS],
    acronym: '',
    name: '',
    pocName: ''
  }
];
