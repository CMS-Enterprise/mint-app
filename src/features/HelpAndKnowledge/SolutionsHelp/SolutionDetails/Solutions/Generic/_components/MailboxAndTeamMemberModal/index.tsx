import React, { useState } from 'react';
import { Trans, useTranslation } from 'react-i18next';
import { Button } from '@trussworks/react-uswds';
import { SolutionContactType } from 'features/HelpAndKnowledge/SolutionsHelp/solutionsMap';

import Modal from 'components/Modal';
import PageHeading from 'components/PageHeading';
import { mtoCommonSolutionContactMisc } from 'i18n/en-US/modelPlan/mtoCommonSolutionContact';

import TeamMailboxForm, { TeamMailboxFormValues } from '../TeamMailboxForm';
import TeamMemberForm, { TeamMemberFormValues } from '../TeamMemberForm';

export type ModeType = TeamMemberModeType | TeamMailboxModeType;
// Matching keys in mtoCommonSolutionContactMisc
export type TeamMemberModeType = 'addTeamMember' | 'editTeamMember';
export type TeamMailboxModeType = 'addTeamMailbox' | 'editTeamMailbox';

const MailboxAndTeamMemberModal = ({
  isOpen,
  closeModal,
  mode,
  contact
}: {
  isOpen: boolean;
  closeModal: () => void;
  mode: ModeType;
  contact?: SolutionContactType;
}) => {
  const { t: contactMiscT } = useTranslation('mtoCommonSolutionContactMisc');

  const [disabledSubmitBtn, setDisableSubmitBtn] = useState(true);

  const [submitTeamMemberForm, setSubmitTeamMemberForm] = useState<
    (formData: TeamMemberFormValues) => void
  >(() => {});

  const [submitTeamMailboxForm, setSubmitTeamMailboxForm] = useState<
    (formData: TeamMailboxFormValues) => void
  >(() => {});

  const isTeamMemberMode =
    mode === 'addTeamMember' || mode === 'editTeamMember';

  const isTeamMailboxMode =
    mode === 'addTeamMailbox' || mode === 'editTeamMailbox';

  return (
    <Modal
      isOpen={isOpen}
      closeModal={closeModal}
      fixed
      className="tablet:width-mobile-lg mint-body-normal overflow-auto"
    >
      <div className="margin-bottom-2">
        <PageHeading headingLevel="h3" className="margin-y-0">
          {mtoCommonSolutionContactMisc[mode].title}
        </PageHeading>
        <p className="text-base margin-y-1">
          <Trans
            i18nKey={mtoCommonSolutionContactMisc.allFieldsRequired}
            components={{
              s: <span className="text-error" />
            }}
          />
        </p>
      </div>

      {isTeamMailboxMode && (
        <TeamMailboxForm
          mode={mode}
          closeModal={closeModal}
          teamMailbox={contact}
          setSubmitForm={setSubmitTeamMailboxForm}
          setDisableButton={setDisableSubmitBtn}
        />
      )}

      {isTeamMemberMode && (
        <TeamMemberForm
          mode={mode}
          closeModal={closeModal}
          teamMember={contact}
          setSubmitForm={setSubmitTeamMemberForm}
          setDisableButton={setDisableSubmitBtn}
        />
      )}

      <div className="margin-top-3 display-flex">
        <Button
          form={isTeamMemberMode ? 'team-member-form' : 'team-mailbox-form'}
          type="submit"
          disabled={disabledSubmitBtn}
          className="margin-right-3 margin-top-0"
          onClick={
            isTeamMemberMode ? submitTeamMemberForm : submitTeamMailboxForm
          }
        >
          {contactMiscT(`${mode}.cta`)}
        </Button>
        <Button
          type="button"
          className="margin-top-0"
          unstyled
          onClick={closeModal}
        >
          {contactMiscT('cancel')}
        </Button>
      </div>
    </Modal>
  );
};

export default MailboxAndTeamMemberModal;
