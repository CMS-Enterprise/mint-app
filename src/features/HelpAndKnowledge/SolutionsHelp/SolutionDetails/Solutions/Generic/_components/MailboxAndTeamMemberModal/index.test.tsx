import React from 'react';
import { MemoryRouter, Route } from 'react-router-dom';
import { MockedProvider } from '@apollo/client/testing';
import { render } from '@testing-library/react';
import { SolutionContactType } from 'features/HelpAndKnowledge/SolutionsHelp/solutionsMap';
import { possibleSolutionsMock } from 'tests/mock/mto';

import MessageProvider from 'contexts/MessageContext';

import MailboxAndTeamMemberModal from '.';

const contact: SolutionContactType = {
  __typename: 'MTOCommonSolutionContact',
  id: '123',
  name: 'Aliza Kim',
  email: 'aliza.kim@cms.hhs.gov',
  mailboxTitle: '',
  mailboxAddress: '',
  isTeam: false,
  role: 'Project Lead',
  isPrimary: true,
  receiveEmails: false
};

const mocks = [...possibleSolutionsMock];

describe('MailboxAndTeamMemberModal Component', () => {
  it('should render add team mailbox context when render', () => {
    const { getByText, queryByText } = render(
      <MemoryRouter
        initialEntries={[
          '/help-and-knowledge/operational-solutions/solutions?solution=accountable-care-organization&section=points-of-contact'
        ]}
      >
        <MockedProvider mocks={mocks} addTypename={false}>
          <MessageProvider>
            <Route path="/help-and-knowledge/operational-solutions">
              <MailboxAndTeamMemberModal
                isOpen
                closeModal={() => {}}
                contact={contact}
                mode="addTeamMailbox"
              />
            </Route>
          </MessageProvider>
        </MockedProvider>
      </MemoryRouter>
    );
    expect(getByText('Add a team mailbox')).toBeInTheDocument();
    expect(queryByText('Add a team member')).not.toBeInTheDocument();
    expect(queryByText('Edit a team member')).not.toBeInTheDocument();
    expect(queryByText('Edit a team mailbox')).not.toBeInTheDocument();
  });

  it('should render add team member context when render', () => {
    const { getByText, queryByText } = render(
      <MemoryRouter
        initialEntries={[
          '/help-and-knowledge/operational-solutions/solutions?solution=accountable-care-organization&section=points-of-contact'
        ]}
      >
        <MockedProvider mocks={mocks} addTypename={false}>
          <MessageProvider>
            <Route path="/help-and-knowledge/operational-solutions">
              <MailboxAndTeamMemberModal
                isOpen
                closeModal={() => {}}
                contact={contact}
                mode="addTeamMember"
              />
            </Route>
          </MessageProvider>
        </MockedProvider>
      </MemoryRouter>
    );
    expect(getByText('Add a team member')).toBeInTheDocument();
    expect(queryByText('Add a team mailbox')).not.toBeInTheDocument();
    expect(queryByText('Edit a team member')).not.toBeInTheDocument();
    expect(queryByText('Edit a team mailbox')).not.toBeInTheDocument();
  });

  it('should render edit team mailbox context when render', () => {
    const { getByText, queryByText } = render(
      <MemoryRouter
        initialEntries={[
          '/help-and-knowledge/operational-solutions/solutions?solution=accountable-care-organization&section=points-of-contact'
        ]}
      >
        <MockedProvider mocks={mocks} addTypename={false}>
          <MessageProvider>
            <Route path="/help-and-knowledge/operational-solutions">
              <MailboxAndTeamMemberModal
                isOpen
                closeModal={() => {}}
                contact={contact}
                mode="editTeamMailbox"
              />
            </Route>
          </MessageProvider>
        </MockedProvider>
      </MemoryRouter>
    );
    expect(getByText('Edit a team mailbox')).toBeInTheDocument();
    expect(queryByText('Edit a team member')).not.toBeInTheDocument();
    expect(queryByText('Add a team mailbox')).not.toBeInTheDocument();
    expect(queryByText('Add a team member')).not.toBeInTheDocument();
  });

  it('should render edit team member context when render', () => {
    const { getByText, queryByText } = render(
      <MemoryRouter
        initialEntries={[
          '/help-and-knowledge/operational-solutions/solutions?solution=accountable-care-organization&section=points-of-contact'
        ]}
      >
        <MockedProvider mocks={mocks} addTypename={false}>
          <MessageProvider>
            <Route path="/help-and-knowledge/operational-solutions">
              <MailboxAndTeamMemberModal
                isOpen
                closeModal={() => {}}
                contact={contact}
                mode="editTeamMember"
              />
            </Route>
          </MessageProvider>
        </MockedProvider>
      </MemoryRouter>
    );
    expect(getByText('Edit a team member')).toBeInTheDocument();
    expect(queryByText('Add a team mailbox')).not.toBeInTheDocument();
    expect(queryByText('Add a team member')).not.toBeInTheDocument();
    expect(queryByText('Edit a team mailbox')).not.toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter
        initialEntries={[
          '/help-and-knowledge/operational-solutions/solutions?solution=accountable-care-organization&section=points-of-contact'
        ]}
      >
        <MockedProvider mocks={mocks} addTypename={false}>
          <MessageProvider>
            <Route path="/help-and-knowledge/operational-solutions">
              <MailboxAndTeamMemberModal
                isOpen
                closeModal={() => {}}
                contact={contact}
                mode="addTeamMember"
              />
            </Route>
          </MessageProvider>
        </MockedProvider>
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
