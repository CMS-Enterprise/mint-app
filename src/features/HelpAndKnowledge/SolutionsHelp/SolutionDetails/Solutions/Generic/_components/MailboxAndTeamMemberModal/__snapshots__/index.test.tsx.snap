// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MailboxAndTeamMemberModal Component > matches snapshot 1`] = `<DocumentFragment />`;
